test/e2e/velero_utils.go

```diff
@@ -109,6 +114,9 @@ func installVeleroServer(io *cliinstall.InstallOptions) error {
 
 	errorMsg := "\n\nError installing Velero. Use `kubectl logs deploy/velero -n velero` to check the deploy logs"
 	resources := install.AllResources(vo)
+	if err = patchResources(io.Namespace, registryCredentialFile, resources); err != nil {
```

Comment:
I know that there's a check inside patchResources to see if registryCredentialFile is empty but it's easier for me to understand if there's also a check here to not call patchResources if registryCredentialFile is empty.  I knew it was possible (probable) so as part of reading the code I needed to go make sure patchResources wouldn't crash if registryCredentialFile was empty.  It's also unclear from here if patchResources might do something even if registryCredentialFile is empty, since we are providing resources.

Reply:
Your comment reminds me that the current logic is incorrect
The `patchResources` not only adds the image pull secret, it also changes the image pull policy, so no matter the `registryCredentialFile` is empty or not, it always changes the image pull policy.

In this case, we can only check the empty inside the `patchResources` method

Reply:
@dsu-igeek I have updated the logic

Reply:
Setting the image pull policy to `Always` causes the E2E testing on Github Action failed, this is because we build the `velero:pr-test` image locally and don't push it into any registry. With image pull policy set to `Always`, the image pull process will fail.

So I remove the setting image pull policy part, and we'll make sure the images are cleaned up before running testing on our regular E2E testing pipeline